Add info for shipping step with downloadable products (#962)
magento/merchdocs: src/catalog/product-create-downloadable.md

- Customers who purchase a downloadable product can be required to log in before going through checkout.

- The delivery of a downloadable product can be made when the order is in either a `Pending` or `Invoiced` status.

> - Since downloadable products are not shipped, If the cart only contains the downloadable product, then the `Shipping` step of the checkout is skipped.

Reviewer: ```suggestion
- Because downloadable products are not shipped, the _Shipping_ step of the checkout is skipped when the cart  contains only the downloadable product.
```
